### ISSUE TITLE
[BUG] Issue when saving menu-item with customfields

### ISSUE THREAD

**Issue by melcarthus** [bacbos/bolt-menu-editor#66]
BOLT 3.3.1
Menu editr 3.2.2
Translate 4.0.9

I have added two extra fields for translation purposes. On saving the menu item I get the fields as aspected.  
![image](https://user-images.githubusercontent.com/7837653/28964183-028df476-790c-11e7-8770-05c8135f1efe.png)

Only after filling the fields, saving and opening the menu-item again I see the default fields are there again together with the extra fields I just saved.

![image](https://user-images.githubusercontent.com/7837653/28964301-8536eca2-790c-11e7-8733-646fe6b2af94.png)

I'm I doing something wrong or is this a bug?

My config looks like this

```yaml
# Backup settings
backups:
    enable: true # Whether to enable backups or not
    folder: 'config://extensions/menu_backups' # This has to be a filesystem path
    keep: 10 # How many backups to keep

# You can define fields in addition to the ones that already exist on menuitems.
# These are key-value pairs of name and default value and will be added (with
# that default value) to all existing items that do not have that field yet.
# You can not use the terms id, nestedSortableItem and nestedSortable-item.

fields:
   Delabel: 'Duitse menu naam'
   Detitle: 'Duitse link titel'

# If addcustomfields is set to false you are not able to add or delete your own fields.
# Only the predefined fields are available then.

# addcustomfields: true

# Add extra options to your menu item fields.
# There are 4 types of fields and the fallback is a text input.
# text: Add a input text field.
# textarea: Add a textarea if you need more chars.
# checkbox: Add a checkbox for lettng the editor easily toggle menu features or so.
# select: Prevent miss spelled values and let the editor just select from a couple of options
# The select needs a property values with key: value pairs.
# Default all fields can be deleted. If you want to avoid the deletion, add delete: false.

menufields:
    main:
        type:
            type: select
            values: {image: Image, something: Just something else}
        image:
            type: text
            delete: false
        path:
            delete: false
        use_icon:
            type: checkbox
        tooltip:
            type: textarea

# Permission for the menu editor
# Use an existing permission from the /bolt/roles page, or create your own
permission: files:config
```

**Comment by melcarthus** [comment]
Nothing is saving correct either. It wil only save 1 menu item correctly. But when I edit another and save the menu again only the last save is saved, and the previous menu-item is reverted back to image nr 2, it loses its settings.

**Comment by anketwokings** [comment]
+1 Kind of the same issue;

**Steps that I take:**

- I edit one field (first item in menu X)

- refresh: Field is reverted back to earlier state
AND: a completely different entry in menu Y (but also the first item of that menu) _is_ changed with the field I edited in menu X...

Really weird behaviour, could you take a look at it? Thanks, and thank you for the extension!
It makes my clients smile.. :)

using: 
Menu Editor 3.2.2
Bolt 3.4.2

**Comment by SahAssar** [comment]
@anketwokings @melcarthus Are both of your configs using uppercase characters for the key? I'm just scanning over it here, but it's not completely off that might be the issue.

**Comment by melcarthus** [comment]
@SahAssar I'll have to dust of this project because i have not been working on it for a while. But will let you know if this is the case soon if nobody else beats me to the punch.

**Comment by anketwokings** [comment]
@jadwigo can you answer this?

> Are both of your configs using uppercase characters for the key?

**Comment by jadwigo** [comment]
We're using lowercase keys .. the menu.yml is the followoing (a bit truncated)

```yml
main:
    -
        label: Traineeopleidingen
        path: landingpage/traineeopleiding
        submenu:
            -
                label: 'Profiel en toelatingseisen'
                path: page/toelating-en-selectie
            -
                label: 'Selectieprocedure 2017'
                path: page/selectiestappen
            -
                label: 'Traineeopleiding Wetgevingsjurist'
                path: page/de-wetgevingsjurist
                options: is-first-sub
                submenu:
                    -
                        label: 'De opleiding'
                        path: page/wetgevingsjurist-opleiding
                    -
                        label: Impressiefilmpjes
                        path: page/wetgevingsjurist-impressiefilmpjes
                        options: link-in-knop
                    -
                        label: 'Meld je aan als geïnteresseerde'
                        path: page/inschrijven_traineeopleiding
                        options: link-in-knop
    -
        label: 'Cursussen en maatwerk'
        path: landingpage/cursusoverzicht
        submenu:
            -
                label: E-learning
                path: page/e-learning
                options: link-in-knop
            -
                label: Maatwerk
                path: page/maatwerk
                options: is-first-sub
                submenu:
                    -
                        label: Werkwijze
                        path: page/werkwijze
                    -
                        label: Locatie
                        path: page/locatie
                    -
                        label: 'Eerder maatwerk'
                        path: page/eerder-maatwerk
                    -
                        label: Internationaal
                        path: landingpage/internationaal
    -
        label: 'Mijn Academie'
        path: page/mijn-academie
        class: hide-for-desktop
    -
        label: Contact
        path: page/contact
        class: hide-for-desktop
    -
        label: English
        path: landingpage/english
        class: hide-for-desktop

header:
    -
        label: 'Mijn Academie'
        path: page/mijn-academie
    -
        label: Contact
        path: page/contact
    -
        label: English
        path: landingpage/english
```

**Comment by anketwokings** [comment]
@melcarthus This is probably fixed in the new version, 3.2.3. Can you confirm?
thx!

**Comment by SahAssar** [comment]
Closing cause fixed, let me know otherwise.
